controllers/componentversion_controller.go

```diff
@@ -154,35 +185,78 @@ func (r *ComponentVersionReconciler) checkVersion(ctx context.Context, obj *v1al
 	return true, latest, nil
 }
 
-func (r *ComponentVersionReconciler) reconcile(ctx context.Context, obj *v1alpha1.ComponentVersion, version string) (ctrl.Result, error) {
+func (r *ComponentVersionReconciler) reconcile(ctx context.Context, obj *v1alpha1.ComponentVersion, version string) (result ctrl.Result, retErr error) {
 	log := log.FromContext(ctx).WithName("ocm-component-version-reconcile")
 
-	// get component version
-	// TODO@souleb: resolve all referenced dependencies and pass them to the function
-	// this will allow us to return early if the dependencies are not ready, and set error
-	// conditions on the component version.
-	// We can the pass the needed dependencies to the function as Option
+	defer func() {
```

Comment:
is there any need for this defer in the inner reconcile method? Could it not be handled in the outer reconcile?